crates/holochain_p2p/src/op_store.rs

```diff
@@ -14,6 +14,42 @@ use kitsune2_api::*;
 use std::collections::HashSet;
 use std::fmt::{Debug, Formatter};
 use std::rc::Rc;
+use std::sync::Arc;
+
+/// Holochain implementation of the Kitsune2 [OpStoreFactory].
+#[derive(Debug)]
+pub struct HolochainOpStoreFactory {
+    /// The database connection.
+    pub db: DbWrite<DbKindDht>,
```

Comment:
Wouldn't there be one of these per space? The factory should be able to create new op stores for any space I think

Reply:
D'oh! - grr... : ( I guess we'll need to pass in some kind of factory function here too then : (

Reply:
Okay, expanded HolochainP2pConfig to take getter callbacks for the databases. The op store indeed needs to be different based on dna hash: https://github.com/holochain/holochain/pull/4732/commits/a2006722ef0ec53550019426da6e264d79e8c966